src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -21,6 +21,7 @@
 #include <algorithm>
 #include <optional>
 #include <random>
+#include <utility>
```

Comment:
nit: stray line?

Reply:
Leftover from a commit where I incorrectly used std::exchange. If there aren’t any major objections I’d leave it here and avoid another CI cycle